SortApp/SortApp.Presentation/SaveArray/SaveArrayService.cs

```diff
@@ -21,6 +21,9 @@ public void ShowSaveDialog(ViewModelBase dataContext)
 				DataContext = dataContext
 			};
 
+			if (((SaveWindowViewModel)dataContext).CloseAction == null)
```

Comment:
Not too good to make such casts.
If you need this thing like that - parameter dataContext should get type SaveWindowViewModel